test/metabase/util/ssh_test.clj

```diff
@@ -198,3 +204,35 @@
           (u/deref-with-timeout server-thread 12000)
           (with-open [in-client (BufferedReader. (InputStreamReader. (.getInputStream socket)))]
             (is (= "hello from the ssh tunnel" (.readLine in-client)))))))))
+
+(deftest test-ssh-tunnel-reconnection
+  (driver/with-driver :postgres
+    (testing "ssh tunnel is reestablished if it becomes closed, so subsequent queries still succeed"
+      (let [real-db-details   (:details (mt/db))
+            tunnel-db-details (assoc real-db-details
+                                     :tunnel-enabled     true
+                                     :tunnel-host        "127.0.0.1"
+                                     :tunnel-auth-option "password"
+                                     ;; trying to use the mock sshd instance doesn't work
+                                     ;; fails with:
+                                     ;;2021-01-28 23:06:04,956 WARN channel.ClientChannelPendingMessagesQueue :: operationComplete(ClientChannelPendingMessagesQueue[channel=TcpipClientChannel[id=0, recipient=-1]-ClientSessionImpl[jsmith@/127.0.0.1:12221], open=false]) SshChannelOpenException[Generic error while opening channel: 0] signaled
+                                     :tunnel-port        ssh-mock-server-with-password-port
```

Comment:
@robdaemon , could use your eyes here.  Basically, I can't make this work with the "mock" ssh server that this test starts up.  Not sure what I'm doing wrong.  I'm basically trying to just use the same params as in `connects-with-correct-password`, but it fails to establish the tunnel in the first place.  If I swap these out for my real ssh port and username/password (provided via OS X), then this test works fine.  Perhaps I'm misunderstanding how to actually use the mock fixtures here.

Reply:
Figured it out.  The local password server needed to have the `AcceptAllForwardingFilter`, like the key server already did.